pyuvdata/mwa_corr_fits.py

```diff
@@ -0,0 +1,485 @@
+# - * - coding: utf-8 - * -
+"""Class for reading MWA correlator FITS files
+
+"""
+from astropy.io import fits
+import numpy as np
+import warnings
+from astropy.time import Time
+from astropy import constants as const
+
+from . import UVData
+from . import utils as uvutils
+
+
+def input_output_mapping():
+    '''
+    Builds a mapping dictionary from the pfb output numbers (the correlator
+    indices for antenna number and polarization) to the pfb input numbers.
+    These input numbers can be mapped to antenna numbers using metadata.
+    '''
+    # this comes from mwa_build_lfiles/mwac_utils.c
+    # inputs are mapped to outputs via pfb_mapper as follows
+    # (from mwa_build_lfiles/antenna_mapping.h):
+    # floor(index/4) + index%4 * 16 = input
+    # for the first 64 outputs, pfb_mapper[output] = input
+    pfb_mapper = [0, 16, 32, 48, 1, 17, 33, 49, 2, 18, 34, 50, 3, 19, 35, 51,
+                  4, 20, 36, 52, 5, 21, 37, 53, 6, 22, 38, 54, 7, 23, 39, 55,
+                  8, 24, 40, 56, 9, 25, 41, 57, 10, 26, 42, 58, 11, 27, 43, 59,
+                  12, 28, 44, 60, 13, 29, 45, 61, 14, 30, 46, 62, 15, 31, 47,
+                  63]
+    # build a mapper for all 256 inputs
+    pfb_inputs_to_outputs = {}
+    for p in range(4):
+        for i in range(64):
+            pfb_inputs_to_outputs[pfb_mapper[i] + p * 64] = p * 64 + i
+    return pfb_inputs_to_outputs
+
+
+class MWACorrFITS(UVData):
+    """
+    Defines a MWA correlator fits-specific subclass of UVData for reading MWA
+    correlator fits files. This class should not be interacted with directly,
+    instead use the read_mwa_corr_fits method on the UVData class.
+    """
+
+    def correct_cable_length(self, cable_lens):
+        """
+        A helper function that applies a cable length correction to the data.
+        """
+        # "the velocity factor of electic fields in RG-6 like coax"
+        # from MWA_Tools/CONV2UVFITS/convutils.h
+        v_factor = 1.204
+        # check if the cable length already has the velocity factor applied
+        cable_array = []
+        for i in cable_lens:
+            if i[0:3] == 'EL_':
+                cable_array.append(float(i[3:]))
+            else:
+                cable_array.append(float(i) * v_factor)
+        # build array of differences
+        cable_len_diffs = np.zeros((self.Nblts, 1))
+        for j in range(self.Nblts):
+            cable_len_diffs[j] = cable_array[self.ant_2_array[j]] - cable_array[self.ant_1_array[j]]
+        self.data_array *= np.exp(-1j * 2 * np.pi * cable_len_diffs / const.c.to('m/s').value
+                                  * self.freq_array.reshape(1, self.Nfreqs))[:, :, None]
+
+    def read_mwa_corr_fits(self, filelist, use_cotter_flags=False, correct_cable_len=True,
+                           phase_data=True, phase_center=None, run_check=True, check_extra=True,
+                           run_check_acceptability=True):
+        """
```

Comment:
I know you were probably following existing code style, but we're in the process of converting to Numpy docstring style (https://numpydoc.readthedocs.io/en/latest/format.html). Can you make your docstrings follow that style?